src/@next/components/organisms/AdyenPaymentGateway/AdyenPaymentGateway.tsx

```diff
@@ -0,0 +1,183 @@
+  errors,
+  onError,
+}: IProps) => {
+  const adyenClientKey = config?.find(({ field }) => field === "client_key")
+    ?.value;
+  const adyenConfig = config?.find(({ field }) => field === "config")?.value;
+  const parsedAdyenConfig = adyenConfig && JSON.parse(adyenConfig);
+
+  const [dropin, setDropin] = useState<any>();
+  const gatewayRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (adyenClientKey && parsedAdyenConfig && !dropin && gatewayRef.current) {
+      const link = document.createElement("link");
+      link.rel = "stylesheet";
+      link.href = styleConfig.src;
+      link.integrity = styleConfig.integrity;
+      link.crossOrigin = styleConfig.crossOrigin;
+      document.body.appendChild(link);
+
+      const script = document.createElement("script");
+      script.src = scriptConfig.src;
+      script.integrity = scriptConfig.integrity;
+      script.crossOrigin = scriptConfig.crossOrigin;
+      script.async = true;
+      script.onload = initAdyenGatewayHandlers; // Wait until the script is loaded before initiating AdyenCheckout
+      document.body.appendChild(script);
+    }
+  }, [adyenClientKey, parsedAdyenConfig, gatewayRef.current]);
+
+  const initAdyenGatewayHandlers = () => {
+    const configuration = adyenClientKey &&
+      adyenConfig && {
+        locale: navigator.language,
+        environment: "test",
+        clientKey: adyenClientKey,
+        paymentMethodsResponse: parsedAdyenConfig,
+        showPayButton: false,
+        onSubmit: (state: any, dropin: any) => {
+          if (!state?.isValid) {
+            onError([new Error("Invalid payment submission")]);
+          } else {
+            submitPayment(state?.data)
+              .then(value => {
+                if (value.error) {
+                  onError([value.error]);
+                } else if (!value?.confirmationNeeded) {
+                  submitPaymentSuccess(value?.order);
+                } else {
+                  const paymentAction =
+                    value?.confirmationData &&
+                    JSON.parse(value?.confirmationData);
+                  dropin.handleAction(paymentAction);
+                }
+              })
+              .catch(error => {
+                onError([new Error(error)]);
+              });
+          }
+        },
+        onAdditionalDetails: (state: any, dropin: any) => {
+          if (!state?.isValid) {
+            onError([new Error("Invalid payment submission")]);
+          } else {
+            submitPayment(state?.data)
+              .then(value => {
+                if (value.error) {
+                  onError([value.error]);
+                } else if (!value?.confirmationNeeded) {
+                  submitPaymentSuccess(value?.order);
+                } else {
+                  const paymentAction =
+                    value?.confirmationData &&
+                    JSON.parse(value?.confirmationData);
+                  dropin.handleAction(paymentAction);
```

Comment:
Same as above
Actually this looks like the same handler, so it can be extracted somewhere